README.rst

```diff
@@ -175,22 +162,8 @@ you can receive all the latest updates, and switch back to version 5.x behavior:
     >>> dt.tzinfo
     <DstTzInfo 'Europe/Vienna' CET+1:00:00 STD>
 
-Version 6 is on `branch master <https://github.com/collective/icalendar/>`_ with compatibility to Python versions ``3.7+`` and ``PyPy3``.
-We expect the ``master`` branch with versions ``6+`` to receive the latest updates and features.
-
-Version 5
-~~~~~~~~~
-
-Version 5 uses only the ``pytz`` timezone implementation, and not ``zoneinfo``.
-No updates will be released for this.
-Please use version 6 and switch to use ``zoneinfo`` as documented above.
-
-Version 4
-~~~~~~~~~
-
-Version 4 is on `branch 4.x <https://github.com/collective/icalendar/tree/4.x>`_ with maximum compatibility with Python versions ``2.7`` and ``3.4+``, ``PyPy2`` and ``PyPy3``.
-The ``4.x`` branch only receives security and bug fixes if someone makes the effort.
-We recommend migrating to later Python versions and also providing feedback if you depend on the ``4.x`` features.
+Version 6 is on `branch main <https://github.com/collective/icalendar/>`_ with compatibility to C-Python and PyPy3.
```

Comment:
We need to be explicit with version numbers of Python and use text. I used the trove classifiers in `setup.py` to get the specific versions. We might want to add 3.12, now that it has been released, but only if it passes tests.

```suggestion
Version 6 is on `branch main <https://github.com/collective/icalendar/>`_.
It is compatible with Python versions 3.7 - 3.11, and PyPy3.
```

Reply:
Since #645, we have a badge to list the versions. I would like to de-duplicate this.
The intention is to support all Python releases that are not end of life. How do you think we can improve this?

Reply:
A badge only shows after rendered through the web. Even then it's not accessible for people with vision impairments and it lacks a context without an HTML `alt` attribute. It's not a satisfactory replacement, but it is a nice visual enhancement for people without vision impairments.

In a README.md, you should have a clear indicator of version support somewhere, whether that is explicit text, a statement to read the trove classifiers in `setup.py`, or a link to read documentation. IMO, it's most common and easiest to keep it in README.md, even though that means more maintenance and risk of duplication mistakes.

In documentation with Sphinx, you can use replacements to avoid maintenance and duplication mistakes. We do this in Plone. Example:

[conf.py](https://github.com/plone/documentation/blob/21453a6ea6c528bbe8bb0799d688d5dc2a5d1809/docs/conf.py#L348-L359)

In the documentation source files:

https://github.com/plone/documentation/blob/21453a6ea6c528bbe8bb0799d688d5dc2a5d1809/docs/install/create-project.md?plain=1#L76-L76

Which renders as https://6.docs.plone.org/install/create-project.html#pre-requisites-for-installation

If you wanted to go fully automatic, I suppose you could change the `setup(app)` to parse trove classifiers and set the result as the value for the replacement variable.

Reply:
> The intention is to support all Python releases that are not end of life.

This should be documented at a policy level, and implemented throughout the repo wherever necessary.

I would go a step further, though, to future proof it. For example:

"icalendar supports all [released and supported versions of Python](https://devguide.python.org/versions/#supported-versions) at the time of its own release."

Thus if you released 6.0 today, then you would support Python 3.8-3.12 only.

In a future release, if you add or drop a version, then that could be a breaking change or a feature addition. You would follow your version numbering scheme accordingly when you make a release at that time.